stage1/init/path.go

```diff
@@ -31,56 +31,56 @@ const (
 	socketsWantsDir = unitsDir + "/sockets.target.wants"
 )
 
-// ServiceUnitName returns a systemd service unit name for the given imageID
-func ServiceUnitName(imageID types.Hash) string {
-	return types.ShortHash(imageID.String()) + ".service"
+// ServiceUnitName returns a systemd service unit name for the given app name.
+func ServiceUnitName(appName types.ACName) string {
+	return appName.String() + ".service"
 }
 
 // ServiceUnitPath returns the path to the systemd service file for the given
-// imageID
-func ServiceUnitPath(root string, imageID types.Hash) string {
-	return filepath.Join(common.Stage1RootfsPath(root), unitsDir, ServiceUnitName(imageID))
+// app name.
+func ServiceUnitPath(root string, appName types.ACName) string {
+	return filepath.Join(common.Stage1RootfsPath(root), unitsDir, ServiceUnitName(appName))
 }
 
-// RelEnvFilePath returns the path to the environment file for the given imageID
-// relative to the pod's root
-func RelEnvFilePath(imageID types.Hash) string {
-	return filepath.Join(envDir, types.ShortHash(imageID.String()))
+// RelEnvFilePath returns the path to the environment file for the given app name
+// relative to the pod's root.
+func RelEnvFilePath(appName types.ACName) string {
+	return filepath.Join(envDir, appName.String())
 }
 
-// EnvFilePath returns the path to the environment file for the given imageID
-func EnvFilePath(root string, imageID types.Hash) string {
-	return filepath.Join(common.Stage1RootfsPath(root), RelEnvFilePath(imageID))
+// EnvFilePath returns the path to the environment file for the given app name.
+func EnvFilePath(root string, appName types.ACName) string {
+	return filepath.Join(common.Stage1RootfsPath(root), RelEnvFilePath(appName))
 }
 
 // ServiceWantPath returns the systemd default.target want symlink path for the
-// given imageID
-func ServiceWantPath(root string, imageID types.Hash) string {
-	return filepath.Join(common.Stage1RootfsPath(root), defaultWantsDir, ServiceUnitName(imageID))
+// given app name.
+func ServiceWantPath(root string, appName types.ACName) string {
+	return filepath.Join(common.Stage1RootfsPath(root), defaultWantsDir, ServiceUnitName(appName))
 }
 
 // InstantiatedPrepareAppUnitName returns the systemd service unit name for prepare-app
-// instantiated for the given root
-func InstantiatedPrepareAppUnitName(imageID types.Hash) string {
+// instantiated for the given root.
+func InstantiatedPrepareAppUnitName(appName types.ACName) string {
 	// Naming respecting escaping rules, see systemd.unit(5) and systemd-escape(1)
-	escaped_root := common.RelAppRootfsPath(imageID)
+	escaped_root := common.RelAppRootfsPath(appName)
 	escaped_root = strings.Replace(escaped_root, "-", "\\x2d", -1)
 	escaped_root = strings.Replace(escaped_root, "/", "-", -1)
```

Comment:
I guess we can remove that since ACName doesn't allow slashes.

Maybe it's good to use go-systemd's escape?